src/game/client/display.cpp

```diff
@@ -74,9 +78,110 @@ void Display::Reset()
  */
 void Display::Update()
```

Comment:
This function needs a checkup by someone else, it seems to work right but im not sure if i didn't miss anything